platform/mbed_critical.c

```diff
@@ -23,6 +23,21 @@
 #include "platform/mbed_assert.h"
 #include "platform/mbed_toolchain.h"
 
+// if __EXCLUSIVE_ACCESS rtx macro not defined, we need to get this via own-set architecture macros
+#ifndef __EXCLUSIVE_ACCESS
+#ifndef MBED_EXCLUSIVE_ACCESS
+#if ((__ARM_ARCH_7M__      == 1U) || \
+    (__ARM_ARCH_7EM__     == 1U) || \
+    (__ARM_ARCH_8M_BASE__ == 1U) || \
+    (__ARM_ARCH_8M_MAIN__ == 1U)) || \
+    (__ARM_ARCH_7A__ == 1U)
+#define MBED_EXCLUSIVE_ACCESS      1U
+#else
+#define MBED_EXCLUSIVE_ACCESS      0U
+#endif
+#endif
```

Comment:
You're not doing anything with `__EXCLUSIVE_ACCESS` here. Did you mean to have an `#else #define MBED_EXCLUSIVE_ACCESS __EXCLUSIVE_ACCCESS`?

Reply:
+1, added new commit addressing it